File: 1456-maximum-number-of-vowels-in-a-substring-of-given-length/1456-maximum-number-of-vowels-in-a-substring-of-given-length.java
class Solution {
    public boolean isVowel(char ch){
        return ch=='a'||ch=='e'||ch=='i'||ch=='o'||ch=='u';
    }
    public int maxVowels(String s, int k) {
       int i=0,j=0; 
        int cntVowel=0;
        for(;j<k;j++)if(isVowel(s.charAt(j)))cntVowel++;
        int windowVowel=cntVowel;
        for(;j<s.length();j++,i++){
            if(isVowel(s.charAt(j)) && isVowel(s.charAt(i))==false){
                windowVowel++;
                cntVowel=Math.max(windowVowel,cntVowel);
            }
            else if(isVowel(s.charAt(j))==false && isVowel(s.charAt(i)))windowVowel--;
        }
        return cntVowel;
    }
}

File: 1456-maximum-number-of-vowels-in-a-substring-of-given-length/1456-maximum-number-of-vowels-in-a-substring-of-given-length.cpp
class Solution {
public:
    bool isVowel(char ch){
        return ch=='a'||ch=='e'||ch=='i'||ch=='o'||ch=='u';
    }
    int maxVowels(string s, int k) {
        
        int i=0,j=0;
        int cntVowels=0;
        for(j=0;j<k;j++)if(isVowel(s[j]))cntVowels++;
        int windowVowels=cntVowels;
        for(j;j<s.size();j++,i++){
            if(isVowel(s[j]) && !isVowel(s[i])){
                windowVowels++;
                cntVowels=max(cntVowels,windowVowels);
            }
            else if(!isVowel(s[j]) && isVowel(s[i]))windowVowels--;
        }
        return cntVowels;
    }
};